test/SpokePool.Deposit.ts

```diff
@@ -402,23 +411,313 @@ describe("SpokePool Depositor Logic", async function () {
+          .connect(depositor)
+          .depositUSS(...getDepositArgsFromRelayData({ ...relayData, inputToken: weth.address }), { value: 0 })
+      ).to.changeTokenBalances(weth, [depositor, spokePool], [amountToDeposit.mul(toBN("-1")), amountToDeposit]);
+    });
+    it("pulls input token from caller", async function () {
+      await expect(() => spokePool.connect(depositor).depositUSS(...depositArgs)).to.changeTokenBalances(
+        erc20,
+        [depositor, spokePool],
+        [amountToDeposit.mul(toBN("-1")), amountToDeposit]
+      );
+    });
+    it("emits USSFundsDeposited event with correct deposit ID", async function () {
+      await expect(spokePool.connect(depositor).depositUSS(...depositArgs))
+        .to.emit(spokePool, "USSFundsDeposited")
+        .withArgs(
+          relayData.inputToken,
+          relayData.outputToken,
+          relayData.inputAmount,
+          relayData.outputAmount,
+          relayData.destinationChainId,
+          // deposit ID is 0 for first deposit
+          0,
+          quoteTimestamp,
+          relayData.fillDeadline,
+          relayData.exclusivityDeadline,
+          relayData.depositor,
+          relayData.recipient,
+          relayData.exclusiveRelayer,
+          relayData.message
+        );
+    });
+    it("deposit ID state variable incremented", async function () {
+      await spokePool.connect(depositor).depositUSS(...depositArgs);
+      expect(await spokePool.numberOfDeposits()).to.equal(1);
+    });
+    it("tokens are always pulled from caller, even if different from specified depositor", async function () {
+      const balanceBefore = await erc20.balanceOf(depositor.address);
+      const newDepositor = randomAddress();
+      await expect(
+        spokePool
+          .connect(depositor)
+          .depositUSS(...getDepositArgsFromRelayData({ ...relayData, depositor: newDepositor }))
+      )
+        .to.emit(spokePool, "USSFundsDeposited")
+        .withArgs(
+          relayData.inputToken,
+          relayData.outputToken,
+          relayData.inputAmount,
+          relayData.outputAmount,
+          relayData.destinationChainId,
+          0,
+          quoteTimestamp,
+          relayData.fillDeadline,
+          relayData.exclusivityDeadline,
+          // New depositor
+          newDepositor,
+          relayData.recipient,
+          relayData.exclusiveRelayer,
+          relayData.message
+        );
+      expect(await erc20.balanceOf(depositor.address)).to.equal(balanceBefore.sub(amountToDeposit));
+    });
+    it("deposits are not paused", async function () {
+      await spokePool.pauseDeposits(true);
+      await expect(spokePool.connect(depositor).depositUSS(...depositArgs)).to.be.revertedWith("Paused deposits");
+    });
+    it("reentrancy protected", async function () {
+      // In this test we create a reentrancy attempt by sending a fill with a recipient contract that calls back into
+      // the spoke pool via the deposit function.
```

Comment:
fwiw I think we should decouple the deposit-side locks from the fill-side locks. Probably not for now, but it could be useful for multi-hop transactions.

Reply:
hmm yeah I can see a reason for that but it would add complexity. Should we try to get into this audit?

Reply:
I think it'd be fine for next cycle - this one is already quite packed.